### PR TITLE
Migrate to using YAML files to store product information

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request is a re-run of #412 with the issues raised (hopefully) resolved. It converts the existing `SOFTWARE-LIST.md` Markdown file into individual YAML files grouped by vendor. When a `push` is made to the `develop` branch the `update_software_list.yml` GitHub Actions workflow will run and update the comprehensive `data/cisagov.yml` file from the separate source YAML files and then generate an updated `SOFTWARE-LIST.md`.

This pull request is in concert with the documentation changes in #425.

## 💭 Motivation and context ##

Using a machine readable file with stricter formatting requirements to store this information should reduce human error and also make it easier for third parties to ingest our data. Using separate YAML files will ensure that pull requests using the GitHub Web UI is a viable option for less technical contributors.

This new method is reliant on the changes introduced in [v1.1.0 of our Python library related to this effort](https://github.com/cisagov/log4j-md-yml/releases/tag/v1.1.0).

## 🧪 Testing ##

Automated tests pass. Functionality testing was made in [this branch of a testing repository](https://github.com/cisagov/github-actions-playground/tree/testing/log4j_list_update).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by Lcerkov — APPROVED** [review]
Everything looks as expected!

File: config/SOFTWARE-LIST.tpl.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+| Under Investigation | Vendor investigating status. |
+
+## Software List ##
+
+This list was initially populated using information from the following sources:
+
+- Kevin Beaumont
+- SwitHak
+
+{{software_markdown_table}}
```
Comment (dav3r) [inline config/SOFTWARE-LIST.tpl.md:23]: I think we should mention something like this:
```suggestion
NOTE: This file is automatically generated.  To make updates, please refer to
[`CONTRIBUTING.md`](CONTRIBUTING.md).

{{software_markdown_table}}
```

**Review by dav3r — APPROVED** [review]
Looks great to me, though I did make one suggestion that I think should be included. 🦖

File: config/SOFTWARE-LIST.tpl.md (hunk 1 above)
Comment (Lcerkov) [reply config/SOFTWARE-LIST.tpl.md:23]: @dav3r Good addition!

Comment (mcdonnnj) [reply config/SOFTWARE-LIST.tpl.md:23]: Added in 3991f5e.

**Pull request merged**